extensions/ql-vscode/src/databaseFetcher.ts

```diff
@@ -45,36 +45,27 @@ async function databaseFetcher(
   await fs.ensureDir(storagePath);
   const unzipPath = await getStorageFolder(storagePath, databaseUrl);
 
-  const response = await fetch.default(databaseUrl);
-  const unzipStream = unzipper.Extract({
-    path: unzipPath
-  });
-  progressCallback({
-    maxStep: 3,
-    message: 'Unzipping database',
-    step: 2
-  });
-  await new Promise((resolve, reject) => {
-    response.body.on('error', reject);
-    unzipStream.on('error', reject);
-    unzipStream.on('close', resolve);
-    response.body.pipe(unzipStream);
-  });
-  progressCallback({
+  if (isFile(databaseUrl)) {
+    await readAndUnzip(databaseUrl, unzipPath);
+  } else {
+    await fetchAndUnzip(databaseUrl, unzipPath, progressCallback);
+  }
+
+  progressCallback?.({
     maxStep: 3,
     message: 'Opening database',
     step: 3
   });
 
-  // if there is a single directory inside, then assume that's what we want to import
-  const dirs = await fs.readdir(unzipPath);
-  const dbPath = dirs?.length === 1 && (await fs.stat(path.join(unzipPath, dirs[0]))).isDirectory
-    ? path.join(unzipPath, dirs[0])
-    : unzipPath;
-
-  // might need to upgrade before importing...
-  const item = await databasesManager.openDatabase(Uri.parse(dbPath));
-  databasesManager.setCurrentDatabaseItem(item);
+  const dbPath = await findDirWithFile(unzipPath, '.dbinfo');
```

Comment:
Note that this is `codeql-database.yml` for newer databases.

Reply:
Yikes. Is there a stable file to look for?

Reply:
Seems like we want to check for the presence of either `codeql-database.yml` or `.dbinfo`, yeah?

Reply:
Checking for the presence of either of those two seems appropriate to me.